cns/configuration/configuration.go

```diff
@@ -35,6 +35,7 @@ type CNSConfig struct {
 	EnableSubnetScarcity        bool
 	EnableSwiftV2               bool
 	InitializeFromCNI           bool
+	EnableHomeAZ                bool
```

Comment:
I think the previous comment got lost but I think we need further discussion on the name of this field. I don't think `EnableHomeAZ` is the correct name because:
- Home AZ is already enabled when mode is direct, which makes this flag confusing.
- The major feature is the NNC creation, which should be independent of home az being available or not.
I still think `EnableNNCCreation` is a better name. Would like to get feedback from @thatmattlong and @rbtr on this.